src/components/Switch/Switch.tsx

```diff
@@ -1,184 +1,231 @@
+                    flexDirection: 'row',
+                    alignItems: 'center',
+                },
+                toggleStyle: [
+                    toggle,
+                    color && { backgroundColor: color },
+                    {
+                        height: size,
+                        width: size * 1.62,
+                        borderRadius: size / 1.62,
+                        borderWidth: Math.floor(size * 0.095),
+                    },
+                    value && { borderWidth: 0 },
+                ],
+                toggleWheelStyle: [
+                    toggleWheel,
+                    {
+                        marginLeft: moveToggle,
+                        width: toggleSize,
+                        height: toggleSize,
+                        borderRadius: (size * 0.84) / 2,
+                    },
+                    thumbColor && { backgroundColor: thumbColor },
+                    !value &&
+                        !unCheckedIcon &&
+                        !pressed && {
+                            borderRadius: (size * 0.5) / 2,
+                        },
+                    pressed && {
+                        borderRadius: (size * (!value ? 1 : 0.85)) / 2,
+                    },
+                ],
+                overlayStyle: [
+                    toggleWheel,
+                    {
+                        marginLeft: moveToggle,
+                    },
+                    thumbOverlay,
+                ],
+                iconStyle: icon,
+            };
+        }, [
+            componentStyles,
+            size,
+            color,
+            moveToggle,
+            toggleSize,
+            thumbColor,
+            value,
+            unCheckedIcon,
+            pressed,
+            thumbOverlay,
+        ]);
 
-export const defaultStyles: ComponentStylePropWithVariants<
-    ViewStyle,
-    'selected_disabled' | 'active' | 'disabled',
-    CustomProp
-> = {
-    thumbTintColor: 'colors.neutral1',
-    onTintColor: 'colors.onNeutral1',
+    const handleValueChange = useCallback(() => {
+        onValueChange && onValueChange(!value);
+    }, [value, onValueChange]);
 
-    states: {
-        active: {
-            thumbTintColor: 'colors.neutral1',
-            onTintColor: 'colors.onNeutral1',
+    const renderIcon = useCallback(
+        (icon: string | undefined, condition: boolean, type?: IconType) => {
+            return icon && condition ? (
+                <Icon style={iconStyle} name={icon} size={size * 0.5} type={type} />
+            ) : null;
         },
-        disabled: {
-            thumbTintColor: 'colors.disabled',
-            onTintColor: 'colors.disabledOnBackground',
-        },
-        selected_disabled: {
-            thumbTintColor: 'colors.neutral1',
-            onTintColor: 'colors.disabledOnBackground',
-        },
-    },
+        [Icon, iconStyle, size],
+    );
+
+    const onIconComponent = useMemo(
```

Comment:
These are redundant.

Below code would just also work without needing unnecessary useCallback and multiple useMemos

{(checkedIcon || unCheckedIcon) ? (
                <Icon style={iconStyle} name={value ? checkedIcon : unCheckedIcon} size={size * 0.5} type={value ? checkedIconType : uncheckedIconType} />
            ) : null}